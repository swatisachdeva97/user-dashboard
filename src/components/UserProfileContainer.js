import {React, Component} from 'react';
import Spinner from 'react-bootstrap/Spinner';
import PropTypes from 'prop-types';
import Service from '../services/Service';


export default class UserProfileContainer extends Component {
    //Initial state
    state = {
        isLoading: false,
        isShowData: false,
        clickedIndex: 0
    };



    onClickHandler = (index)=> {
        this.setState({
            isShowData: false,
            isLoading: true,
            clickedIndex: index
        });

        setTimeout(()=> {
            this.setState({
                isLoading: false,
                isShowData: true,
                clickedIndex: index
            })
        }, 2000)
    };

    render() {
        let {userData} = this.props;
        return (
            <div className="container outer-container">
                <div className="row">
                    {
                        userData.map((data, index)=>(
                            <div key={index}>

                                <div className="inner-container">
                                    <div className="column">
                                        <img className="avatar" src={data.avatar}/>
                                    </div>
                                    <div>
                                        <a className="user-data" onClick={(e)=> this.onClickHandler(index)}>

                                            <div className="first-name"> Name: {data.first_name} {data.last_name}</div>
                                            <div className="state">State: {data.address.state}</div>
                                            <div className="country">Country: {data.address.country}</div>
                                            <div className="phone">Phone: { data.phone_number}</div>
                                            <div className="email">Email: { data.email}</div>
                                        </a>
                                    </div>
                                    <div className="cd-folding-panel">
                                        <div className="additional-details">
                                            {this.state.isLoading && this.state.clickedIndex == index ?
                                                <Spinner animation="border"/> : null}
                                            {this.state.isShowData && this.state.clickedIndex == index ?
                                                <div className="show-data-fold-out">
                                                    <div>Title: {data.employment.title}</div>
                                                    <div>Key Skill: {data.employment.key_skill}</div>
                                                    <div>Gender: {data.gender}</div>
                                                </div>
                                                :
                                                null
                                            }
                                        </div>
                                    </div>
                                </div>
                            </div>
                        ))
                    }
                </div>
            </div>
        )
    }
}

UserProfileContainer.propTypes = {
    userData: PropTypes.array
};