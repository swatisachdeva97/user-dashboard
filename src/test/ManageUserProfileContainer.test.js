import { render, screen } from '@testing-library/react';
import {shallow, mount} from 'enzyme';
import toJson from "enzyme-to-json";
import ManageUserProfileContainers from './../components/ManageUserProfileContainer';
import UserProfileContainer from './../components/UserProfileContainer';
import { configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
configure({ adapter: new Adapter() });

describe('<ManageUserProfileContainers>', ()=>{

    it('matches snapshot', () =>{
        const wrapper = shallow(<ManageUserProfileContainers/>);
        expect(toJson(wrapper)).toMatchSnapshot()
    });

   it('should find div', () =>{
        const wrapper = shallow(<ManageUserProfileContainers/>);
       expect(wrapper.find("div").length).toEqual(1);
    });

    it('verify initial states', () =>{
        const wrapper = shallow(<ManageUserProfileContainers/>);
        expect(wrapper.state().userData.length).toEqual(0);
        expect(wrapper.state().filteredData.length).toEqual(0);
    });

    it('verify updated state', () =>{
        const wrapper = shallow(<ManageUserProfileContainers/>);

        wrapper.state().userData = [
            {
                address: {
                    city: "Hamillhaven",
                    street_name: "Feest Field",
                    street_address: "4589 Adrian Land",
                    zip_code: "30531-0927",
                    state: "Hawaii"
                },
                avatar: "https://robohash.org/itaqueutsimilique.png?size=300x300&set=set1",
                date_of_birth: "1997-12-15",
                email: "homer.smith@email.com",
                employment: {title: "Lead Education Coordinator", key_skill: "Problem solving"},
                first_name: "Homer",
                gender: "Bigender",
                id: 2948,
                last_name: "Smith",
                phone_number: "+965 1-915-048-1041",
                uid: "38973318-7d18-4660-a4cc-a1c0c21bde80",
            }
        ];
        expect(wrapper.state().userData.length).toEqual(1);
        expect(wrapper.state().filteredData.length).toEqual(0);
    });


});
