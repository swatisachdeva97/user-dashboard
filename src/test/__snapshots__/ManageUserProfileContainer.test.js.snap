// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ManageUserProfileContainers> matches snapshot 1`] = `
<div>
  <UserProfileContainer
    userData={Array []}
  />
</div>
`;
